user/super/com/google/gwt/emul/java/util/Optional.java

```diff
@@ -96,6 +105,22 @@ public <U> Optional<U> flatMap(Function<? super T, Optional<U>> mapper) {
     return empty();
   }
 
+  public Optional<T> or(Supplier<? extends Optional<? extends T>> supplier) {
```

Comment:
Good catch, will add.

Reply:
@niloc132 the first null check is implemented, the second one still missing (compared to https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/master/src/java.base/share/classes/java/util/Optional.java#L312 )

Reply:
Unless the source is apache licensed, please don't link to it (or otherwise use it to author our own implementations of java interfaces) - prefer Javadoc, or tests of the current implementation behavior for that. From context, I'm guessing that the supplier cannot return null (since I am not going to click, and I can't imagine what else could reasonably need a null check).

Reply:
In my first comment I quoted JavaDoc, which states that NPE should also be thrown if the supplier returns null. That's why we should add two not null checks.

Reply:
Thanks, I missed the "or". Corrected this, and added tests for both cases.